--- DistributedTransactionsDist-source/src/main/java/agh/sr/dtransactions/DBTxConfiguration.java	2014-03-24 02:26:52.388755247 +0100
+++ DistributedTransactionsDist/src/main/java/agh/sr/dtransactions/DBTxConfiguration.java	2014-03-24 02:28:30.328756069 +0100
@@ -43,6 +43,8 @@
 
 	private static final int NUMBER_OF_USERS = 10;
 
+	private static String OS = System.getProperty("os.name").toLowerCase();
+
 	private static String[][] CUSTOMERS = new String[][] {
 			{ "Marek", "Kowalski", "Krakow" }, { "Jacek", "Nowak", "Wroclaw" },
 			{ "Maciek", "Zielinski", "Poznan" },
@@ -69,12 +71,45 @@
 		}
 	}
 
+	public static boolean isWindows() {
+
+		return (OS.indexOf("win") >= 0);
+
+	}
+
+	public static boolean isMac() {
+
+		return (OS.indexOf("mac") >= 0);
+
+	}
+
+	public static boolean isUnix() {
+
+		return (OS.indexOf("nix") >= 0 || OS.indexOf("nux") >= 0 || OS
+				.indexOf("aix") > 0);
+
+	}
+
+	public static boolean isSolaris() {
+
+		return (OS.indexOf("sunos") >= 0);
+
+	}
+
 	private static Connection ensureThatDBisCreated(String name,
 			String dbSchemaFile) throws Exception {
 		String dbUrl = DERBY_URL_PREFIX + name;
 		File dbDirectory = new File(name);
 		if (!dbDirectory.exists()) {
-			Runtime.getRuntime().exec("cmd.exe /c rmdir /q /s " + name);
+			if (isWindows()) {
+				Runtime.getRuntime().exec("cmd.exe /c rmdir /q /s " + name);
+			} else if (isUnix()) {
+				Runtime.getRuntime().exec("rm " + name);
+			}
+			else {
+				throw new RuntimeException("Invalid os");
+			}
+
 			Class.forName("org.apache.derby.jdbc.EmbeddedDriver");
 			Connection dbConnection = DriverManager.getConnection(dbUrl
 					+ ";create=true");
@@ -226,4 +261,3 @@
 	}
 
 }
-
